pluma/pluma-view.c

```diff
@@ -694,18 +709,36 @@ pluma_override_font (const gchar          *item,
 				pango_font_description_get_size (font) / PANGO_SCALE,
 				pango_font_description_get_size_is_absolute (font) ? "px" : "pt");
 
-	provider = gtk_css_provider_get_default ();
+	if (!provider_added)
+		provider = gtk_css_provider_new ();
 
 	prov_str = gtk_css_provider_to_string (provider);
 
 	if (g_str_has_prefix (prov_str, "textview") && g_str_has_prefix (item, ".context-menu"))
-		css = g_strdup_printf ("%s %s { %s %s %s %s }", prov_str, item, family, weight, style, size);
+	{
+		if (strstr (prov_str, ".context-menu"))
+		{
+			g_strdelimit (prov_str, "}", '\0');
+			gchar *prov_new_str = g_strdup_printf ("%s}", prov_str);
+			css = g_strdup_printf ("%s %s { %s %s %s %s }", prov_new_str, item, family, weight, style, size);
+			g_free (prov_new_str);
+		}
+		else
+			css = g_strdup_printf ("%s %s { %s %s %s %s }", prov_str, item, family, weight, style, size);
+	}
 	else
 		css = g_strdup_printf ("%s { %s %s %s %s }", item, family, weight, style, size);
 
 	gtk_css_provider_load_from_data (provider, css, -1, NULL);
 
-	if (!provider_added) {
+	if (!provider_added)
+	{
+		static GSettings *settings; /*We need this for the life of the provider and program*/
+		settings = g_settings_new ("org.mate.interface");
+		g_signal_connect (settings,
+				  "changed::" "font-name",
+				  G_CALLBACK (contextmenu_font_changed_cb), NULL);
```

Comment:
Here you can probably pass `widget` instead of `NULL`, that way you get the right object in the callback when doing `GTK_WIDGET (user_data)`.